### ISSUE TITLE
speechbrain.nnet.loss.transducer_loss.py vs torchaudio.transforms.RNNTLoss

### ISSUE THREAD

**Issue by kokamido** [speechbrain/speechbrain#1100]
Hi! In PyTorch 1.10 the [RNNTLoss was added](https://pytorch.org/blog/pytorch-1.10-new-library-releases/#beta-rnn-transducer-loss). Will you maintain the speechbrain [transducer_loss](https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/nnet/losses.py#L27) in the future? Should I prefer PyTorch RNNTLoss over it?

**Comment by TParcollet** [comment]
Hi, we are on the process of benchmarking both @aheba . It is highly probable that we will maintain both in the end, as the torch loss won't enable users to do research on this loss, while our implementation does.

**Comment by kokamido** [comment]
Thank you for the answer. Will you publish the results of your benchmarking? It's interesting for me because PyTorch release notes describe the implementation as following: 

> The implementation is consistent with the original loss function in  Sequence Transduction with Recurrent Neural Networks, but relies on code from Alignment Restricted Streaming Recurrent Neural Network Transducer.

In the second paper the authors pay particular attention to the token emission delays and propose the modified RNN-t loss called "Alignment Restricted RNN-T Loss". So PyTorch implementation may perform significantly different than the standard RNN-T Loss from the first paper.

**Comment by TParcollet** [comment]
There is no reason for us not making the benchmark public @aheba. But this is still on the to-do list, so we do not know when it will be up :-)

**Comment by aheba** [comment]
Hi @kokamido , good point,
Let me open a PR to support RNN-T torchaudio.

**Comment by aheba** [comment]
Hi @kokamido , The PR  #1199  is merged, Know we do support both RNN-T loss from torchaudio and our.

In next days, I will push a PR on adding:
1- FastEmit Transducer loss regularisation for more streamable Transducer model
2- Work on decoding part with a stateless decoding

**Comment by csukuangfj** [comment]
> Hi @kokamido , The PR #1199 is merged, Know we do support both RNN-T loss from torchaudio and our.
> 
> In next days, I will push a PR on adding: 1- FastEmit Transducer loss regularisation for more streamable Transducer model 2- Work on decoding part with a stateless decoding

@aheba 
Do you plan to make the benchmarks public?
